src/SFA.DAS.EmployerApprenticeshipsService.Application.UnitTests/Services/ProviderEmailNotificationServiceTests/WhenSendingProviderCommitmentStopNotification.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Moq;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using SFA.DAS.Commitments.Api.Types.Apprenticeship;
+using SFA.DAS.EmployerCommitments.Application.Services;
+using SFA.DAS.EmployerCommitments.Domain.Interfaces;
+using SFA.DAS.EmployerCommitments.Domain.Models.Notification;
+using SFA.DAS.HashingService;
+using SFA.DAS.NLog.Logger;
+
+namespace SFA.DAS.EmployerCommitments.Application.UnitTests.Services.ProviderEmailNotificationServiceTests
+{
+    public class WhenSendingProviderCommitmentStopNotification
+    {
+        private ProviderEmailNotificationService _providerEmailNotificationService;
+
+        private Mock<IProviderEmailService> _providerEmailService;
+        private Mock<IHashingService> _hashingService;
+
+        private Apprenticeship _exampleApprenticeship;
+
+        private EmailMessage _sentEmailMessage;
+
+        private Func<Task> _act;
+
+        [SetUp]
+        public void Arrange()
+        {
+            _providerEmailService = new Mock<IProviderEmailService>();
+            _providerEmailService.Setup(x =>
+                x.SendEmailToAllProviderRecipients(It.IsAny<long>(), It.IsAny<string>(), It.IsAny<EmailMessage>()))
+                .Callback<long, string, EmailMessage>((l, s, m) => _sentEmailMessage = m)
+                .Returns(Task.CompletedTask);
+
+            _hashingService = new Mock<IHashingService>();
+            _hashingService.Setup(x => x.HashValue(It.IsAny<long>())).Returns("HASH");
+
+            _providerEmailNotificationService =
+                new ProviderEmailNotificationService(_providerEmailService.Object, Mock.Of<ILog>(), _hashingService.Object);
+
+            var payload = new Apprenticeship
+            {
+                ProviderId = 123L,
+                LegalEntityName = "Legal Entity Name",
+                ULN = "1234567890",
+                FirstName = "John",
+                LastName = "Smith",
+                StopDate = new DateTime(2018, 05, 01)
+            };
+            _exampleApprenticeship =
```

Comment:
Why the serialization round-trip?
If we want to test if the serialization round-trip is successful, it should be in its own test.